rfcs/0000-unified-configuration.ftd

```diff
@@ -0,0 +1,188 @@
+-- import: fastn.com/rfcs/lib
+-- import: fastn-community.github.io/bling/note
+
+-- lib.rfc: RFC: Unified Configuration
+id: unified-config
+status: proposal
+
+TODO: Write a brief summary of the RFC here.
+
+
+-- lib.motivation:
+
+;; TODO: add links to the following paragraph
+Currently, `fastn` supports configuration through environment variables, and in
+some way, through the `FASTN.ftd` file. Environment variables are good to
+change the behaviour of the cli or to pass secret information but it's not
+ideal for defining complex configuration for a framework like `fastn` that
+provides various configurable features like processors, oauth authentication
+and, apps.
+
+;; TODO: I can do better than this
+This RFC proposes a way of dealing with configuration that is centralised and
+leveraged the existing ability of the `fastn` cli to read environment
+variables.
+
+-- end: lib.motivation
+
+
+-- lib.detailed-design:
+
+-- ds.h2: `fastn/config` - A virtual module to assist configuration definition.
+
+Add a virtual module (`fastn/config`) similar to existing [assets](TODO: add
+link) module.
+
+;; TODO add footer section for explanors like foreign variables and virtual module
+- This module contains all special configuration types defined by the fastn
+  framework. This module also provides safe access to environment variables
+  through foreign variables.
+
+- `FASTN.ftd` file will be central to the whole configuration system. Users can
```

Comment:
We have to mention `FASTN.secrets.ftd` also here.

Reply:
I am leaving `FASTN.secrets.ftd` as I believe that it can be a separate RFC (or simply a PR). 
For context, `FASTN.secrets.ftd` is just an alternative to `.env` file.